### ISSUE TITLE
Normal user permissions

### ISSUE THREAD

**Issue by zonedgopher** [HeyuX10Automation/heyu#23]
Now that we have the heyu upload issue in version 2.11-rc3 patched, I need to open another issue regarding user permissions.
When heyu is installed, during the "make install" process, the install changes the permissions on two directories:
1.  /var/tmp/heyu is chmod to 777
2. /var/lock is chmod to 1777 (sticky bit).
This allows a normal user to execute heyu commands and works fine.
However, once you reboot the Linux machine, the re-boot process automatically changes the permissions back to root access only.  Thus, after re-boot the normal user can no longer execute commands like "heyu upload", only root or elevated user can execute those commands.
Is possible to correct this so normal user can always have access, even after re-boot?

**Comment by jkrzyszt** [comment]
Heyu is not responsible for setting file permissions, it just requires them to be set correctly.  You should be able to resolve your issue by editing your system startup scripts.  If needed, you can also ask for help on heyu_users Yahoo group.

Thanks,
Janusz

**Comment by keithehenry** [comment]
Zoned, look in the Configure, or Configure.sh file for -DLOCKDIR. I copied 
`-DLOCKDIR=\"/var/spool/locks\"'`
from freebsd (or other) $SYS decode into the linux case and it seems to lock and work between reboots, as desired. I have no idea if this is the correct fix or not but you might give it a try.

**Comment by zonedgopher** [comment]
You lost me on this fix.  I found the line you referred to in Configure.  What are you saying to do with it?  Modify something else in heyu install, or put it somewhere in Linux?  Not sure what $SYS and "decode" refer to.
Thanks for the note.

**Comment by zonedgopher** [comment]
Oh.  I think I figured it out.  You copied that line in configure to the section that refers to Linux - right?
I will try that.
Thanks for the tip

**Comment by keithehenry** [comment]
I see now that we could also use `sh ./Configure.sh openbsd` (or several other options) when running Configure.sh to set LOCKDIR. It doesn't appear that it would have any other effect. That way we wouldn't have to modify any code. [If not otherwise defined, LOCKDIR appears to be set to /var/lock in X10.h.]

**Comment by zonedgopher** [comment]
Thanks for the additional tip.  I had not tried the 1st suggestion because on my fedora there is not a directory /var/spool/locks.  So, I wasn't sure how to adjust the script.
My lock files are in /var/lock (actually /var/run/lock).  So this might work.
Thanks for the update and additional info.

**Comment by zonedgopher** [comment]
I tried installing with the command --> sh  ./Configure.sh  linux
The install worked and everything worked fine in heyu.
However, when I re-booted, the permissions on /var/lock were changed back to 755, and normal user could not run heyu commands.
Maybe I did this wrong or misunderstood you message? or maybe it just won't work on my fedora 29 install.
It was worth trying.
Thanks

**Comment by keithehenry** [comment]
I am suggesting that /var/lock aka /var/run/lock is not working for us. Therefore, I used the /var/spool folder which is used by several other linux-unix-like systems and it worked. I am suggesting that perhaps it doesn't make any difference where we put the lock file. The Configure.sh for generic linux systems isn't working for fedora nor arch-linux (mine) but is configurable and parameterized.  Hint: use `sh ./Configure.sh openbsd` as I stated before.

**Comment by heyu-users** [comment]
On Fedora, /usr/lib/tmpfiles.d is a directory of configuration files 
that get run during the boot process. The file named legacy.conf clears 
several lock directories including the /run/lock directory and resets 
the permissions to 755. It then creates symbolic link /var/lock to 
/run/lock.

When heyu is installed it changes the lock directory permissions to 777 
allowing everyone write access. After a reboot the legacy.conf script 
runs and sets the lock directory permissions to 755 preventing any 
further non-root write access.

One method to fix this is to edit /usr/lib/tmpfiles.d/legacy.conf and 
change the line with:

d /run/lock 0755 root root -

to:

d /run/lock 0777 root root -

Alternatively, changing the heyu source to use a different lock 
directory like /var/spool/lock works because after heyu creates the 
directory as world writable, no other process is aware of it so it won't 
be changed by anything other than heyu.

Fair winds,
Mike
nz9p

**Comment by zonedgopher** [comment]
Mike,
Thanks for your very clear and concise explanation.
I appreciate it very much.

**Comment by zonedgopher** [comment]
Keith,
as an update,
The command sh ./Configure.sh openbsd will install heyu on Fedora29, but heyu will not run.  It throws a bunch of errors.
However, as you suggested, if you edit the "Linux" section of Configure.sh to use the lock directory /var/spool/locks,  and then install with  sh  ./Configure.sh linux, everything works fine.
Thanks for the suggestions.

**Comment by zonedgopher** [comment]
One more comment for anyone that might find it useful:
I chose to modify the configure.sh script to use /var/spool/locks instead of modifying the legacy.conf file in Fedora to use permission 777 on /var/run/lock, because of possible security issues related to permissions on /var/run/lock.
I have been confused by the various discussions I have read about the /var/run/lock permissions issue, because some find concern with 777 and others do not.  There is an old Redhat bug report # 581884 that explains the issue (latent privilege escalation vulnerability), but I am still not sure what is safe, or if this is even a security issue at all.  (The privilege escalation vulnerability is latent because there was no known exploit when the bug was filed in 2010. )
So, I opted for what I think might be the more secure method, leaving permissions on /var/run/lock as fedora sets them, and creating a new lock, /var/spool/locks, that nothing else uses except heyu.
